core/lib/instrument.c

```diff
@@ -62,6 +63,9 @@
 #ifdef UNIX
 #    include <sys/time.h>       /* ITIMER_* */
 #    include "../unix/module.h" /* redirect_* functions */
+#    include "../unix/os_exports.h"
```

Comment:
os_exports.h comes from globals.h deliberately as there are certain ordering requirements in including some of these headers (it also avoids other files from having to add ifdefs like here; though the include dirs can also avoid that).  Adding it here has no effect as it's already included from globals.h.  What is the motivation for this change?  Is it a clang-tidy complaint?  (Good to put that in the PR description.) The model of having os_exports.h obtained from globals.h is baked into every file in core/.  If we want to change that I think we need to make a conscious decision to do it and we'd have to figure out how to then impose the ordering we need and we'd change all the files.  That may not be a trivial project and we'd have to debate whether it makes the code "cleaner" and is worth the effort.

An alternative to avoid clang-tidy in other build systems (xref #4103) would be to put `// IWYU pragma: export` on the includes in globals.h.  That seems much easier and simpler short-term.

(The other option is to do nothing and just ignore these if raised in another system.)

Reply:
I've removed os_exports.h. The current change addresses most of the clang-tidy complaints.

Reply:
> os_exports.h comes from globals.h deliberately as there are certain ordering requirements in including some of these headers

For such ordering requirements, could we fail explicitly in pre-processing using `#error` if the intended ordering is violated? Otherwise it's hard to track down what went wrong.

This can be addressed in a separate PR of course.

Reply:
Let's address it in a separate PR.